homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -13,22 +16,33 @@
     BinarySensorEntity,
 )
 from homeassistant.const import CONF_BINARY_SENSORS, CONF_NAME
-from homeassistant.core import callback
+from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 from homeassistant.util import Throttle
 
 from .const import (
     BINARY_SENSOR_SCAN_INTERVAL_SECS,
     DATA_AMCREST,
     DEVICES,
-    SENSOR_DEVICE_CLASS,
-    SENSOR_EVENT_CODE,
-    SENSOR_NAME,
     SERVICE_EVENT,
     SERVICE_UPDATE,
 )
 from .helpers import log_update_error, service_signal
 
+if TYPE_CHECKING:
+    from . import AmcrestDevice
+
+
+class _SensorParamsType(NamedTuple):
+    """Parameters to define sensors."""
```

Comment:
If this is refactored, it should be refactored to use `SensorEntityDescription` instead.

Reply:
With the current change, this is just adding some type information to the existing `tuple`. In #53683, it seemed like the additional refactoring that was done on top of adding some of the typing caused too many changes to start taking place. I think this makes sense to do, but would say making that sort of change is a good follow-up PR to this one. Especially since there is the event code in this tuple which is a parameter that is used to query the API of the sensor, but isn't necessarily something that fits in that struct. I will follow up with a PR with this change, but how and where to store that might get in the way of this change which is focused on getting all of the typing setup.

Reply:
Not forgetting the fact that #53683 was closed to avoid discussions about testing.

Reply:
I have a commit https://github.com/flacjacket/core/commit/7157913ca15866ac8707ad6f1f6c48b784742610 ready that I can use for a follow-up PR, but that will need some extra testing to be sure it is still functioning correctly.

Reply:
In that case, revert the refactoring done in this PR. If one refactors anything, it should be following the wished-for patterns, not new intermediate ones.

Reply:
Sure, I've spun this back so these are back in their dict form rather than changing that information to a namedtuple. Some type information is needed on the dict, but it is otherwise the same as it was before.